ChronoAPI/ChronoLog/include/RPCVisor.h

```diff
@@ -0,0 +1,432 @@
+//
+// Created by kfeng on 12/26/22.
+//
+
+#ifndef CHRONOLOG_RPCVISOR_H
+#define CHRONOLOG_RPCVISOR_H
+
+#include <iostream>
+#include <unordered_map>
+#include <functional>
+#include <thallium.hpp>
+#include <sys/types.h>
+#include <unistd.h>
+#include "macro.h"
+#include "errcode.h"
+#include "RPCFactory.h"
+#include "ClientRegistryManager.h"
+#include "ChronicleMetaDirectory.h"
+
+extern std::shared_ptr<ClientRegistryManager> g_clientRegistryManager;
+extern std::shared_ptr<ChronicleMetaDirectory> g_chronicleMetaDirectory;
+
+class RPCVisor {
+public:
+    RPCVisor() {
+        LOGD("%s constructor is called", typeid(*this).name());
+        rpc = ChronoLog::Singleton<ChronoLogRPCFactory>::GetInstance()
+                ->GetRPC(CHRONOLOG_CONF->RPC_BASE_SERVER_PORT);
+        set_prefix("ChronoLog");
+        if (CHRONOLOG_CONF->IS_SERVER) {
+            chronicleMetaDirectory = ChronoLog::Singleton<ChronicleMetaDirectory>::GetInstance();
+        }
+        LOGD("%s constructor finishes, object created@%p in thread PID=%d",
+             typeid(*this).name(), this, getpid());
+    }
+
+    ~RPCVisor() = default;
+
+    /**
+     * Admin APIs
+     */
+    int LocalConnect(const std::string &uri, std::string &client_id, int &flags, uint64_t &clock_offset) {
+        LOGD("%s in ChronoLogAdminRPCProxy@%p called in PID=%d, with args: uri=%s",
+             __FUNCTION__, this, getpid(), uri.c_str());
+        ClientInfo record;
+        record.addr_ = "127.0.0.1";
+        if (std::strtol(client_id.c_str(), nullptr, 10) < 0) {
+            LOGE("client id is invalid");
+            return CL_ERR_INVALID_ARG;
+        }
+        if (g_clientRegistryManager) {
+            return g_clientRegistryManager->add_client_record(client_id, record);
```

Comment:
Make this function consistent with LocalDisconnect. Add a return type for all conditions (if-else), otherwise you get compilation warnings.